rust_src/src/symbols.rs

```diff
@@ -58,3 +59,21 @@ fn symbol_function(object: LispObject) -> LispObject {
 fn symbol_plist(object: LispObject) -> LispObject {
     object.as_symbol_or_error().get_plist()
 }
+
+/// Intern a string (e.g. create a symbol from it, if it does not exist
+/// already).
+///
+/// This is an implementation equivalent of the C version.
+pub unsafe fn intern(s: *const libc::c_char) -> LispObject {
+    LispObject::from_raw(intern_1(s, libc::strlen(s) as libc::ptrdiff_t))
+}
+
+/// Intern (e.g. create a symbol for) a `&'static str`, avoiding
+/// allocation. This macro is inherently unsafe: if the string contains internal
+/// NULL bytes (`\0`), we will violate memory safety.
```

Comment:
Let's assume intern_1 didn't take the length; then as far as the C API is concerned the string would end at the first null byte. That's not undefined behavior, but of course not what you intended either.

The method on CString is unsafe because it maintains the invariant that its string has the same length for Rust and C consumers, so not checking must be an unsafe function.